src/libcore/time.rs

```diff
@@ -31,6 +31,29 @@ const NANOS_PER_MICRO: u32 = 1_000;
 const MILLIS_PER_SEC: u64 = 1_000;
 const MICROS_PER_SEC: u64 = 1_000_000;
 
+/// 1 nanosecond `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const NS: Duration = Duration::from_nanos(1);
+/// 1 microsecond `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const US: Duration = Duration::from_micros(1);
+#[unstable(feature = "time_units", issue = "0")]
+/// 1 millisecond `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const MS: Duration = Duration::from_millis(1);
+/// 1 second `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const S: Duration = Duration::from_secs(1);
+/// 1 minute `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const M: Duration = Duration::from_secs(60);
+/// 1 hour `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const H: Duration = Duration::from_secs(60*60);
+/// 1 day `Duration`
+#[unstable(feature = "time_units", issue = "0")]
+const D: Duration = Duration::from_secs(24*60*60);
```

Comment:
Similar to #47097, I doubt we want to support  hours and days. 

Abbreviating minutes to `M` is also confusing (months? meters?). The SI standard short form `MIN` conflicts in meaning with "minimum". I would like to leave this out as well, keeping only `S`, `MS`, `US` and `NS`.